tofhir-server/src/main/scala/io/tofhir/server/service/SchemaDefinitionService.scala

```diff
@@ -124,4 +128,67 @@ class SchemaDefinitionService(schemaRepository: ISchemaRepository) extends LazyL
     JArray(rootElement +: elements)
   }
 
+  /**
+   * Infer the schema
+   * @param inferTask infer schema task object
+   * @return
+   */
+  def inferSchema(inferTask: InferTask): Future[Option[SchemaDefinition]] = {
+
+    // Execute SQL and get the dataFrame
+    val dataFrame = SourceHandler.readSource("unnamed", SparkConfig.sparkSession,
```

Comment:
Does this read the all data or only column metadata ? I think we should add a configuration to read only column metadata for this functionality ?

Reply:
I did not find any option that provides this functionality.

Reply:
From the answers to this SO question (https://stackoverflow.com/questions/44277019/how-to-read-only-n-rows-of-large-csv-file-on-hdfs-using-spark-csv-package), I understand that if we read the data as text and limit, and then apply csv reader on the result, we can achieve what we want during the schema infer process. Can you please check? @Okanmercan99